### DISCUSSION TITLE
MockRender function input

### DISCUSSION THREAD

**Discussion by cheng93** [help-me-mom/ng-mocks#6511] [Q&A] [answered]
```typescript
const funcInput = () => '123';
fixture = MockRender(TestComponent, {
  funcInput
});
```

I believe `funcInput` gets autowired to be an output.

Is it possible to pass a function input in MockRender

**Comment by satanTime** [comment, marked as answer]
Hi @cheng93,

that's not true. `funcInput` will be set as `@Input` or `@Output` of `TestComponent` only when `TestComponent` has such an input or output.

If `TestComponent` has neither, then nothing will be done with `funcInput`.
if `TestComponent` has `@Input() funcInput`, then `this.funcInput` will be the function.
if `TestComponent` has `@Output() funcInput`, then `funcInput` will be triggered on emits of `this.funcInput`.

**Reply by cheng93** [reply]
https://stackblitz.com/edit/github-htgnz6?file=src%2Ftest.spec.ts

any idea why my test is failing then?

**Reply by satanTime** [reply]
Aha, I see, that makes sense, because inputs are wrapped.
Could you tell me why you need to assert that `MockRender` is passing parameters to components correctly?

**Reply by cheng93** [reply]
Ah I think my example was too simple, I also can't remember exactly what I was doing that caused me to ask this question.

I think it was along the following.

https://stackblitz.com/edit/github-htgnz6-ouv9us?file=src%2Ftest.spec.ts
